src/world/action/jump.c

```diff
@@ -1,15 +1,220 @@
+    playerStatus->unk_4C = playerStatus->position.y;
 
-INCLUDE_ASM(void, "world/action/jump", func_802B6294_E242D4, void);
+    phys_init_integrator_for_current_state();
+    collisionStatus = &gCollisionStatus;
 
-INCLUDE_ASM(void, "world/action/jump", func_802B6348_E24388, void);
+    if (playerStatus->animFlags & PLAYER_ANIM_FLAG_8BIT_MARIO) {
+        temp_v1 = 0x90005;
+    }
+    else {
+        if (!(playerStatus->animFlags & (PLAYER_ANIM_FLAG_HOLDING_ITEM | PLAYER_ANIM_FLAG_2))) {
+            temp_v1 = 0x10007;
+        } else {
+            temp_v1 = 0x60009;
+        }
+    }
+    suggest_player_anim_clearUnkFlag(temp_v1);
+    collisionStatus->lastTouchedFloor = collisionStatus->currentFloor;
+    collisionStatus->currentFloor = -1;
+}
 
-INCLUDE_ASM(void, "world/action/jump", func_802B647C_E244BC, void);
+void func_802B60B4_E240F4() {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    s32 phi_a0;
 
-INCLUDE_ASM(void, "world/action/jump", func_802B6508_E24548, void);
+    if (playerStatus->flags < 0) {
+        playerStatus->flags &= (~0x80000000);
+        func_802B6000_E24040();
+
+        if (playerStatus->actionState == ACTION_STATE_LAUNCH) {
+            phys_adjust_cam_on_landing();
+        } else {
+            gCameras[0].moveFlags |= 1;
+        }
+
+        if (playerStatus->actionState == ACTION_STATE_JUMP) {
+            if (playerStatus->animFlags & PLAYER_ANIM_FLAG_8BIT_MARIO) {
+                sfx_play_sound_at_player(SOUND_JUMP_8BIT_MARIO, 0);
+            }
+            else {
+                sfx_play_sound_at_player(SOUND_JUMP_2081, 0);
+            }
+        }
+    }
+
+    if (playerStatus->animFlags & PLAYER_ANIM_FLAG_8BIT_MARIO) {
+        phi_a0 = 0x90005;
+    } else {
+        if (!(playerStatus->animFlags & (PLAYER_ANIM_FLAG_HOLDING_ITEM | PLAYER_ANIM_FLAG_2))) {
+          phi_a0 = 0x10007;
+        }
+        else {
+            phi_a0 = 0x60009;
+        }
+    }
+
+    suggest_player_anim_clearUnkFlag(phi_a0);
+    playerStatus->decorationList++;
+}
+
+void func_802B6198_E241D8() {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    CollisionStatus* collisionStatus = &gCollisionStatus;
+    Entity* entity;
+    s32 temp_s1;
+    s32 phi_a0;
+
+    temp_s1 = 0x80000000;
+    if (playerStatus->flags < 0) {
+        entity = get_entity_by_index(collisionStatus->currentFloor);
+        D_8010C960 = entity->position.x;
+        D_8010C97C = entity->position.z;
+        func_802B6000_E24040();
+        playerStatus->flags |= 0x880000;
+        disable_player_input();
+    }
+
+    playerStatus->decorationList++;
+    if (playerStatus->fallState == 1) {
+        if (playerStatus->flags & 0x80000000) {
+
+          playerStatus->flags &= 0x7FFFFFF5;
+          playerStatus->flags |= 4;
+
+          phi_a0 = 0x6000A;
+
+          if (!(playerStatus->animFlags & (PLAYER_ANIM_FLAG_HOLDING_ITEM | PLAYER_ANIM_FLAG_2))) {
+            phi_a0 = 0x10008;
+          }
+
+          temp_s1 = phi_a0;
```

Comment:
are you able to just use phi_a0 here and remove temp_s1 entirely?

Reply:
No I tried removing the temp_s1 in this function and replacing it with phi_a0 but it causes reg alloc issues and adds a mov where an nop should be.